Document/0x04f-Testing-Network-Communication.md

```diff
@@ -205,67 +205,123 @@ If you want to verify whether your server supports the right cipher suites, ther
+| **Layer 7 (Application)** | API hooking (`HttpUrlConnection`, `NSURLSession`, `WebRequest`) | Frida | Modifies how apps handle network requests. |
+| **Layer 6 (Presentation)** | Hooking TLS functions (`SSL_read`, `SSL_write`) | Frida, SSL Kill Switch | Intercepts encrypted data before it reaches the app. |
+| **Layer 5 (Session)** | Proxy interception | Burp Suite, ZAP, mitmproxy | Requires app to respect proxy settings. |
+| **Layer 4 (Transport)** | Packet sniffing | `tcpdump`, Wireshark | Captures **all** TCP/UDP traffic but does **not** decrypt HTTPS. |
+| **Layer 3 (Network)** | MITM via ARP spoofing | bettercap | ARP (Layer 2) is used to manipulate **Layer 3** (IP traffic). |
+| **Layer 2 (Data Link)** | Rogue Wi-Fi AP | `hostapd`, `dnsmasq`, `iptables`, `wpa_supplicant`, `airmon-ng` | Captures **all** traffic from a connected device. |
+
+## MASTG-TECH: Intercepting HTTP Traffic Using an Interception Proxy
+
+Interception proxies are the most common method for intercepting mobile app traffic. They work by setting up a proxy server that intercepts and logs all HTTP/HTTPS traffic between the mobile app and the server. This allows you to view and modify the requests and responses in real-time.
+
+Several free and commercial proxy tools are available. For example: @MASTG-TOOL-0077 and @MASTG-TOOL-0079.
+
+### Rerouting Traffic to the Proxy
 
 To use the interception proxy, you'll need to run it on your host computer and configure the mobile app to route HTTP(S) requests to your proxy. In most cases, it is enough to set a system-wide proxy in the network settings of the mobile device - if the app uses standard HTTP APIs or popular libraries such as `okhttp`, it will automatically use the system settings.
 
 <img src="Images/Chapters/0x04f/BURP.png" width="100%" />
 
-Using a proxy breaks SSL certificate verification and the app will usually fail to initiate TLS connections. To work around this issue, you can install your proxy's CA certificate on the device. We'll explain how to do this in the OS-specific "Basic Security Testing" chapters.
+### Installing the Proxy Certificate
 
-## Intercepting Non-HTTP Traffic
+Using an interception proxy breaks SSL certificate verification and the app will usually fail to initiate TLS connections. So, interception proxies require you to install a custom CA certificate on the mobile device, which allows the proxy to decrypt and inspect the encrypted HTTPS traffic. However, some apps implement certificate pinning to prevent this, which requires additional steps to bypass.
+
+### Per-Platform Instructions
+
+- Android: see @MASTG-TECH-0011
+- iOS: see @MASTG-TECH-0063
+
+## MASTG-TECH: Intercepting Non-HTTP Traffic Using an Interception Proxy
 
 Interception proxies such as @MASTG-TOOL-0077 and @MASTG-TOOL-0079 won't show non-HTTP traffic, because they aren't capable of decoding it properly by default. There are, however, Burp plugins available such as:
```

Comment:
```suggestion
Interception proxies such as @MASTG-TOOL-0077 and @MASTG-TOOL-0079 won't show non-HTTP traffic, because they aren't capable of decoding it properly by default. They can, however, be extended using the following tools, allowing you to intercept and manipulate non-HTTP traffic:
```